pkg/webhook/cspc.go

```diff
@@ -413,12 +414,17 @@ func validateBlockDevice(bd *openebsapis.BlockDevice, hostName string) error {
 			bd.Labels[types.HostNameLabelKey],
 		)
 	}
+
+	// If the BD tag is present on BD and the value is empty then
+	// this BD is not a valid BD for provisioning.
 	if v, found := bd.Labels[types.BlockDeviceTagLabelKey]; found {
-		return errors.Errorf(
-			"block device %s is tagged with a value %s and cannot be used",
-			bd.Name,
-			v,
-		)
+		if strings.TrimSpace(v) == "" {
```

Comment:
Should we still deny from webhook if the BD has a tag?

Reply:
I am looking into how we can get rid of the webhook for most of the validations. (Reason: We faced a timeout issue ) Also, there can be a possibility that some improvements can happen in the webhook with regard to timeouts, and if that  is  the case -- putting validation in the webhook helps in two way :
1. It is an instant failure and lookup will not be required by the admins to make the CSPC right
2. Failures if uncorrected -- can cause a lot of logs to get accumulated due to reconcile loop -- but if rejected in webhook -- this can be avoided. 

Had other validations in webhook -- so took a decision to put in there too until we finalise a solution.

Reply:
Was this your question?

Reply:
If BD has a tag and the tag has an empty value -- cStor rejects it.
cStor accepts BD that are not tagged or BD with tags that should be allowed via annotation on CSPC.

Reply:
got it.